great_expectations/profile/json_schema_profiler.py

```diff
@@ -46,10 +46,10 @@ class JsonSchemaProfiler(Profiler):
         JsonSchemaTypes.BOOLEAN.value: ProfilerTypeMapping.BOOLEAN_TYPE_NAMES,
     }
 
-    def __init__(self, configuration: dict = None) -> None:
+    def __init__(self, configuration: Optional[dict] = None) -> None:
         super().__init__(configuration)
 
-    def validate(self, schema: dict) -> bool:
+    def validate(self, schema: dict) -> bool:  # type: ignore
```

Comment:
Could we use `type: ignore[specific-mypy-error-code]` and a comment why we're excluding?

Reply:
Yes!
I didn't even know `mypy` supported this.
https://mypy.readthedocs.io/en/stable/error_codes.html#error-codes

Reply:
@cdkini 
Looks like we can even get `mypy` to enforce this.
https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-type-ignore-include-an-error-code-ignore-without-code

But for some reason, I couldn't get it to work. I will revisit enforcement of this on the next typing PR